### PR TITLE
e2e: Improve waiting for drpc readiness

### LINKED ISSUE #1659: Disable DR after relocate times out randomly
I see this error randomly both locally and in the CI, in the last days we see lot of failures, so this may be a regression in ramen.

```
2024-11-20T19:29:09.997+0200	INFO	appset-deploy-rbd-busybox	dractions/retry.go:115	drpc phase is Relocated
2024-11-20T19:29:09.999+0200	INFO	appset-deploy-rbd-busybox	dractions/retry.go:58	drpc is ready
=== RUN   TestSuites/Exhaustive/appset-deploy-rbd-busybox/Disable
2024-11-20T19:29:09.999+0200	INFO	appset-deploy-rbd-busybox	dractions/actions.go:102	Unprotecting workload
2024-11-20T19:29:09.999+0200	INFO	appset-deploy-rbd-busybox	dractions/actions.go:107	Deleting drpc
    actions_test.go:64: drpc "appset-deploy-rbd-busybox" is not deleted yet before timeout, fail
=== NAME  TestSuites/Exhaustive/appset-deploy-rbd-busybox
    exhaustive_suite_test.go:91: Disable failed
...
--- FAIL: TestSuites (0.05s)
    ...
    --- FAIL: TestSuites/Exhaustive (6.06s)
        ...
        --- FAIL: TestSuites/Exhaustive/appset-deploy-rbd-busybox (1231.36s)
            --- PASS: TestSuites/Exhaustive/appset-deploy-rbd-busybox/Deploy (0.17s)
            --- PASS: TestSuites/Exhaustive/appset-deploy-rbd-busybox/Enable (90.15s)
            --- PASS: TestSuites/Exhaustive/appset-deploy-rbd-busybox/Failover (270.25s)
            --- PASS: TestSuites/Exhaustive/appset-deploy-rbd-busybox/Relocate (270.29s)
            --- FAIL: TestSuites/Exhaustive/appset-deploy-rbd-busybox/Disable (600.49s)
```

Looking at the cluster we see:

On the hub, the drpc is deleting state (good):

```
% kubectl get drpc -A  --context hub                                          
NAMESPACE   NAME                        AGE   PREFERREDCLUSTER   FAILOVERCLUSTER   DESIREDSTATE   CURRENTSTATE
argocd      appset-deploy-rbd-busybox   31m   dr1                dr2               Relocate       Deleting
```

On dr1, the application is running (good), and the vrg/vr were deleted (good):

```
% kubectl get deploy,pod,pvc,vrg,vr -n appset-deploy-rbd-busybox --context dr1
NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/busybox   1/1     1            1           13m

NAME                           READY   STATUS    RESTARTS   AGE
pod/busybox-7d5747dcf9-47b5t   1/1     Running   0          13m

NAME                                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      VOLUMEATTRIBUTESCLASS   AGE
persistentvolumeclaim/busybox-pvc   Bound    pvc-ff343fdd-0a8d-4b2a-90bc-f05e831a11e2   1Gi        RWO            rook-ceph-block   <unset>                 15m
```

On dr2, we see that the application did not complete the cleanup after relocate:

```
% kubectl get deploy,pod,pvc,vrg,vr -n appset-deploy-rbd-busybox --context dr2
NAME                                STATUS        VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      VOLUMEATTRIBUTESCLASS   AGE
persistentvolumeclaim/busybox-pvc   Terminating   pvc-ff343fdd-0a8d-4b2a-90bc-f05e831a11e2   1Gi        RWO            rook-ceph-block   <unset>                 24m

NAME                                                                    DESIREDSTATE   CURRENTSTATE
volumereplicationgroup.ramendr.openshift.io/appset-deploy-rbd-busybox   secondary      Secondary

NAME                                                             AGE   VOLUMEREPLICATIONCLASS   PVCNAME       DESIREDSTATE   CURRENTSTATE
volumereplication.replication.storage.openshift.io/busybox-pvc   24m   vrc-sample               busybox-pvc   secondary      Secondary
```

I think the issue is not waiting for stable state before disabling dr. It may be a bug in ramen that we cannot handle disable dr when relocated did not complete but there is no reason to test this edge case. We need to test the normal case which is:

1. Wait until application is relocated
2. Wait for first replication - meaning that the application is protected on the new cluster
3. Delete the drpc

This is the flow in basic-test, and it works reliably:
https://github.com/RamenDR/ramen/blob/8c376ffb502123d28953bf432923e9b3914ace32/test/basic-test/relocate#L21

In e2e we wait until phase is relocated:
https://github.com/RamenDR/ramen/blob/8c376ffb502123d28953bf432923e9b3914ace32/e2e/dractions/actions.go#L154

This is very early in relocate flow. Relocated did not succeed yet.

Logs from the failing run:
[disable-timeout.gather.tar.gz](https://github.com/user-attachments/files/17834927/disable-timeout.gather.tar.gz)

Failed builds:
- https://github.com/RamenDR/ramen/actions/runs/11940648215/job/33283766375
- https://github.com/RamenDR/ramen/actions/runs/11955746683/job/33336392815
- https://github.com/RamenDR/ramen/actions/runs/11978974524/job/33400129258
- https://github.com/RamenDR/ramen/actions/runs/11996849328/job/33441787711
- https://github.com/RamenDR/ramen/actions/runs/11988822417/job/33424523294
- https://github.com/RamenDR/ramen/actions/runs/11980488460/job/33404837532

DFBUG: https://issues.redhat.com/browse/DFBUGS-969

### PR DESCRIPTION
**Pull request by nirs**
Improve the implementation and logging when waiting until drpc is ready.

- Improve waitDRPCReady logging and error handling
- Unify waiting for failover or relocate

Part of #1686

### CODE REVIEW

**Comment by nirs** [comment]
Marking as draft since it rebased on other PR.

File: e2e/dractions/retry.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/ramendr/ramen/e2e/types"
 	"github.com/ramendr/ramen/e2e/util"
 	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (nirs) [inline e2e/dractions/retry.go:16]: Looks like we can remove one of the imports.

Comment (nirs) [reply e2e/dractions/retry.go:16]: Checked again, we need both.

**Review by raghavendra-talur — APPROVED** [review]
